app/uk/gov/hmrc/eoricommoncomponent/frontend/controllers/subscription/CompanyRegisteredCountryController.scala

```diff
@@ -54,34 +64,53 @@ class CompanyRegisteredCountryController @Inject() (
     }
 
   private def populateView(country: Option[CompanyRegisteredCountry], service: Service, isInReviewMode: Boolean)(
-    implicit request: Request[_]
+    implicit request: Request[AnyContent]
   ): Result = {
 
-    val form = country.fold(CompanyRegisteredCountry.form())(CompanyRegisteredCountry.form().fill(_))
+    val form       = CompanyRegisteredCountry.form(errorMessageBasedOnType)
+    val filledForm = country.fold(form)(form.fill(_))
+
+    Ok(prepareViewBasedOnType(filledForm, service, isInReviewMode))
+  }
 
+  private def prepareViewBasedOnType(form: Form[CompanyRegisteredCountry], service: Service, isInReviewMode: Boolean)(
+    implicit request: Request[AnyContent]
+  ): Html = {
     val (countries, picker) = Countries.getCountryParametersForAllCountries()
 
-    Ok(countryPage(form, countries, picker, service, isInReviewMode))
+    if (requestSessionData.userSubscriptionFlow == MigrationEoriRowOrganisationSubscriptionUtrNinoEnabledFlow)
```

Comment:
can you rename it to organisation and remove nino from the name

Reply:
There is already something with name: `MigrationEoriRowOrganisationSubscriptionFlow`

So I'm not sure about changing this, because change to it will be totally different format than the others options.

In general we can easily change the name of the case object without affecting anything, so it's not a problem. Let's discuss this again.

I think it's worth to review all the subscription flow options and remove not used one

Reply:
I'll refactor this in separate PR, I will take a look at all SubscriptionFlow modes.